<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>SeqPlots - An interactive tool for visualizing NGS signals and sequence 
motif densities along genomic features using average plots and heatmaps. — seqplots • seqplots</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">seqplots</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QuickStart.html">SeqPlots R workflow</a>
    </li>
    <li>
      <a href="../articles/SeqPlotsGUI.html">SeqPlots R workflow</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/przemol/seqplots">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>SeqPlots - An interactive tool for visualizing NGS signals and sequence 
motif densities along genomic features using average plots and heatmaps.</h1>
    </div>

    
    <p>SeqPlots is a tool for plotting next generation sequencing (NGS) based 
experiments' signal tracks, e.g. reads coverage from ChIP-seq, RNA-seq 
and DNA accessibility assays like DNase-seq and MNase-seq, over user 
specified genomic features, e.g. promoters, gene bodies, etc. It can 
also calculate sequence motif density profiles from reference genome. 
The data are visualized as average signal profile plot, with error estimates 
(standard error and 95<!-- % confidence interval) shown as fields, or as series  -->
of heatmaps that can be sorted and clustered using hierarchical clustering, 
k-means algorithm and self organising maps. Plots can be prepared using R 
programming language or web browser based graphical user interface (GUI) 
implemented using Shiny framework. The dual-purpose implementation allows 
running the software locally on desktop or deploying it on server. 
SeqPlots is useful for both for exploratory data analyses and preparing 
replicable, publication quality plots. Other features of the software 
include collaboration and data sharing capabilities, as well as ability 
to store pre-calculated result matrixes, that combine many sequencing 
experiments and in-silico generated tracks with multiple different features. 
These binaries can be further used to generate new combination plots on fly, 
run automated batch operations or share with colleagues, who can adjust 
their plotting parameters without loading actual tracks and recalculating 
numeric values. SeqPlots relays on Bioconductor packages, mainly on 
rtracklayer for data input and BSgenome packages for reference genome 
sequence and annotations.</p>
    

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Useful links:</p><ul>
<li><p>Project web page: <a href='https://github.com/Przemol/seqplots'>https://github.com/Przemol/seqplots</a></p></li>
<li><p>Online documentation: <a href='http://przemol.github.io/seqplots/'>http://przemol.github.io/seqplots/</a></p></li>
<li><p>Project wiki: <a href='https://github.com/przemol/seqplots/wiki'>https://github.com/przemol/seqplots/wiki</a></p></li>
<li><p>Issue tracker: <a href='http://github.com/przemol/seqplots/issues'>http://github.com/przemol/seqplots/issues</a></p></li>
<li><p>Bug reports: <a href='http://github.com/przemol/seqplots/issues/new'>http://github.com/przemol/seqplots/issues/new</a></p></li>
</ul>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p><strong>R project and Bioconductor:</strong></p><ul>
<li><p>R Core Team (2014). R: A language and environment for statistical
   computing. R Foundation for Statistical Computing, Vienna, Austria. URL
   <a href='http://www.R-project.org/'>http://www.R-project.org/</a>.</p></li>
<li><p>Bioconductor: Open software development for
   computational biology and bioinformatics R. Gentleman, V. J. Carey, D. M.
   Bates, B.Bolstad, M.Dettling, S. Dudoit, B. Ellis, L. Gautier, Y. Ge, and
   others 2004, Genome Biology, Vol. 5, R80.  URL
   <a href='http://www.bioconductor.org/'>http://www.bioconductor.org/</a>.</p></li>
<li><p>RStudio and Inc. (2014). shiny: Web
   Application Framework for R. R package version 0.10.1.
   <a href='http://shiny.rstudio.com/'>http://shiny.rstudio.com/</a>.</p></li>
<li><p>Other CRAN packages: digest,
   DBI,RSQLite, jsonlite, plotrix, fields, grid, kohonen, Cairo, parallel</p></li>
<li><p>Bioconductor packages: IRanges, BSgenome, Rsamtools, rtracklayer,
   GenomicRanges and Biostrings</p></li>
</ul><p><strong>JavaScript and CSS</strong></p><ul>
<li><p>jQuery framework - <a href='http://jquery.com'>http://jquery.com</a></p></li>
<li><p>Bootstrap - <a href='http://getbootstrap.com'>http://getbootstrap.com</a></p></li>
<li><p>DataTables, Table plug-in for jQuery - 
   <a href='http://www.datatables.net'>http://www.datatables.net</a></p></li>
<li><p>jQuery File Upload Plugin - 
   <a href='https://github.com/blueimp/jQuery-File-Upload'>https://github.com/blueimp/jQuery-File-Upload</a></p></li>
<li><p>jQuery throttle - 
   <a href='http://benalman.com/projects/jquery-throttle-debounce-plugin/'>http://benalman.com/projects/jquery-throttle-debounce-plugin/</a></p></li>
<li><p>jQuery Cookie Plugin - 
   <a href='https://github.com/carhartl/jquery-cookie'>https://github.com/carhartl/jquery-cookie</a></p></li>
<li><p>Modernizer JS library - <a href='http://modernizr.com'>http://modernizr.com</a></p></li>
<li><p>JavaScript Templates - 
   <a href='https://github.com/blueimp/JavaScript-Templates'>https://github.com/blueimp/JavaScript-Templates</a></p></li>
<li><p>JavaScript Color Picker - <a href='http://jscolor.com'>http://jscolor.com</a></p></li>
<li><p>md5-js - <a href='https://github.com/wbond/md5-js'>https://github.com/wbond/md5-js</a></p></li>
<li><p>Font Awesome - <a href='http://fortawesome.github.io/Font-Awesome'>http://fortawesome.github.io/Font-Awesome</a></p></li>
<li><p>Google Fonts - <a href='https://www.google.com/fonts'>https://www.google.com/fonts</a></p></li>
<li><p>jQuery user interface - <a href='http://jqueryui.com'>http://jqueryui.com</a> (documentation)</p></li>
<li><p>jquery.tocify.js: jQuery Table of Contents - 
   <a href='https://github.com/gfranko/jquery.tocify.js'>https://github.com/gfranko/jquery.tocify.js</a> (documentation)</p></li>
<li><p>Strapdown <a href='https://github.com/arturadib/strapdown'>https://github.com/arturadib/strapdown</a> 
   (documentation)</p></li>
<li><p>Bootswatch themes - <a href='http://bootswatch.com'>http://bootswatch.com</a> (documentation)</p></li>
<li><p>google-code-prettify - 
   <a href='https://code.google.com/p/google-code-prettify'>https://code.google.com/p/google-code-prettify</a> (documentation)</p></li>
<li><p>marked - <a href='https://github.com/chjj/marked'>https://github.com/chjj/marked</a> (documentation)</p></li>
</ul><p><strong>Important conceptual contribution to the project</strong></p><ul>
<li><p>Liu T, Ortiz J, Taing L, Meyer C, Lee B, Zhang Y, Shin H, Wong S, Ma J,
Lei Y, et al. 2011. 
<a href='http://www.ncbi.nlm.nih.gov/pubmed/21859476'>Cistrome: an integrative 
platform for transcriptional regulation studies.</a> Genome Biology 12: R83.</p></li>
<li><p>Thomas Williams, Colin Kelley and others (2010).
Gnuplot 4.4: an interactive plotting program. URL 
<a href='http://www.R-project.org/'>http://www.R-project.org/</a>.</p></li>
<li><p>Kent, W.J., Sugnet, C.W., Furey, T.S., Roskin, K.M., Pringle, T.H.,
Zahler, A.M. and Haussler, a. D. (2002). 
<a href='http://www.ncbi.nlm.nih.gov/pubmed/12045153'>The Human Genome Browser 
at UCSC</a>. Genome Research. 12:996-1006.</p></li>
<li><p>Kent WJ, Zweig AS, Barber G, Hinrichs AS, Karolchik D.
(2010). <a href='http://www.ncbi.nlm.nih.gov/pubmed/20639541'>BigWig and BigBed:
 enabling browsing of large distributed datasets.</a> Bioinformatics.
1;26(17):2204-7</p></li>
<li><p>Nicol, J.W., Helt, G.A., Blanchard, S.G., Raja, A. and Loraine, A.E. 
(2009). <a href='http://www.ncbi.nlm.nih.gov/pubmed/19654113'>The Integrated 
Genome Browser: free software for distribution and exploration of 
genome-scale datasets.</a> Bioinformatics (Oxford, England). 25:2730-1.</p></li>
<li><p>Thorvaldsdottir, H., Robinson, J.T. and Mesirov, J.P. (2012). 
<a href='http://www.ncbi.nlm.nih.gov/pubmed/22517427'>Integrative Genomics 
Viewer (IGV): high-performance genomics data visualization and
exploration.</a> Briefings in bioinformatics. bbs017</p></li>
</ul><p><strong>Server deployment</strong></p><ul>
<li><p>Shiny Server - <a href='https://github.com/rstudio/shiny-server'>https://github.com/rstudio/shiny-server</a></p></li>
<li><p>ShinyApps - <a href='https://github.com/rstudio/shinyapps'>https://github.com/rstudio/shinyapps</a></p></li>
</ul><p><strong>Publications containing figures made by SeqPlots</strong></p><ul>
<li><p>Chen RA, Stempor P, Down TA, Zeiser E, Feuer SK, Ahringer J. 
  <a href='http://www.ncbi.nlm.nih.gov/pubmed/24653213'>Extreme HOT regions are 
  CpG-dense promoters in C. elegans and humans.</a> Genome Res 
  24(7):1138-1146 Jul 2014</p></li>
</ul>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>
          </ul>

    <h2>Author</h2>
    
Przemyslaw Stempor

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
