<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SeqPlots R workflow • seqplots</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">seqplots</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStart.html">SeqPlots R workflow</a>
    </li>
    <li>
      <a href="../articles/SeqPlotsGUI.html">SeqPlots R workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/przemol/seqplots">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>SeqPlots R workflow</h1>
                        <h4 class="author">Przemyslaw Setmpor</h4>
            
            <h4 class="date">02 August 2017</h4>
          </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>SeqPlots is a tool for plotting average track signals (e.g. read coverage) and sequence motif densities over user specified genomic features. The data can be visualized in linear plots with error estimates or as series of heatmaps that can be sorted and clustered. The software can be run locally on a desktop or deployed on a server and allows easy data sharing. SeqPlots pre-calculates and stores binary result matrices, allowing rapid plot generation. Plots can also be run in batch.</p>
    </div>
    
<div class="contents">
<div id="quick-start-demo" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start-demo" class="anchor"></a>Quick start demo</h1>
<ol style="list-style-type: decimal">
<li>
<p>Start SeqPlots. Refer to <strong>installation guides</strong> for platform specific information. After successful initiation the web interface should automatically open in your default web browser. If you are using web server version just navigate your browser to the server address.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_01.png" alt="The SeqPlots interface in web browser"><p class="caption">The SeqPlots interface in web browser</p>
</div>
</li>
<li>
<p>Upload feature (BED or GFF) and track (BigWig or WIG) files. They can be gzip compressed (e.g. file1.bed.gz). Press green “Add files…” button or just drag and drop files into the window. The ready to upload files will show up in upload window, where you select user name, reference genome and optionally add some comments.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_02.png" alt="File upload panel"><p class="caption">File upload panel</p>
</div>
</li>
<li>
<p>When all is done press blue “Start upload” button. After upload and processing is done the green “SUCCESS” label should show. It means that file is on the registered and ready to use. Occasionally the file might be mot formatted properly or chromosome names might not agree with reference genome. In such case a verbose error will window appears and file as labeled as “ERROR”. For further information please refer to <a href="Errors"><strong>errors chapter</strong></a>.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_03.png" alt="File upload progress infoermation"><p class="caption">File upload progress infoermation</p>
</div>
</li>
<li>
<p>Dismiss upload window and press blue “New plot set” button on side panel. This will bring up file management window. In file management window select at least one file from “Features” tab and at least one file from “Tracks” or sequence motif(s). The sequence motifs and tracks can be processed and plotted together. Select files by clicking on file name, selected files will be highlighted.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_04.png" alt="File management panel"><p class="caption">File management panel</p>
</div>
</li>
<li>
<p>After choosing files/motifs to plot, set up the processing options. You can find these in the button of plotting window.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_05.png" alt="Plot set calculation options"><p class="caption">Plot set calculation options</p>
</div>
</li>
<li>
<p>After options are set up press blue “Run calculation” button. This will dismiss the file management window and show processing message. Here you can observe the progress of the task and optionally cancel it if no longer required or you forgot to add some important file to the plot-set.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_06.png" alt="Plot set calculation progress window"><p class="caption">Plot set calculation progress window</p>
</div>
</li>
<li>
<p>After some time the calculation will finish (fingers crossed, without the error) and you will be able to see plot set array. In here you can choose which feature-track or feature-motif pairs to plot. Choose one or more checkboxes and press grey “Line plot” button (or hit RETURN from your keyboard).</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_07.png" alt="The plot selection grid"><p class="caption">The plot selection grid</p>
</div>
</li>
<li>
<p>Congratulation! Your First plot is complete, you can see the preview of it on the side panel.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_08.png" alt="The plot preview panel"><p class="caption">The plot preview panel</p>
</div>
</li>
<li>
<p>You are able to set up labels, titles, font sizes, legends and many more on control panel tabs - please see <strong>Plotting</strong> chapter for details.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_09.png" alt="Plot settings tabs"><p class="caption">Plot settings tabs</p>
</div>
</li>
<li>
<p>By clicking the plot preview you can enlarge it for better view. When everything is ready you can get the plot as PDF by clicking green “Line plot” button just on the top of side panel.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_10.png" alt="Average plot example"><p class="caption">Average plot example</p>
</div>
</li>
<li>
<p>You can also visualize the signal as a heatmap. Please note that heatmap plotting is possible only for a single feature file. It is possible to sort heatmaps based on mean row signals and/or cluster them using k-means algorithm, hierarchical clustering or self organising maps. To learn more about heatmaps see <strong>Heatmaps</strong> chapter.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_11.png" alt="Heatmap settings tab"><p class="caption">Heatmap settings tab</p>
</div>
</li>
<li>
<p>Heatmaps can be downloaded sa PDFs using ‘Heatmap’ button just on the bottom of setup panel. The small button, at the right sied of ‘Heatmap’ button downloads cluster definition and sorting order.</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_12.png" alt="Heatmap example"><p class="caption">Heatmap example</p>
</div>
</li>
</ol>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<pre><code>## R version 3.4.0 (2017-04-21)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.5
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] compiler_3.4.0  backports_1.1.0 magrittr_1.5    rprojroot_1.2  
##  [5] tools_3.4.0     htmltools_0.3.6 yaml_2.1.14     Rcpp_0.12.12   
##  [9] stringi_1.1.5   rmarkdown_1.6   knitr_1.16      stringr_1.2.0  
## [13] digest_0.6.12   evaluate_0.10.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#quick-start-demo">Quick start demo</a></li>
      <li><a href="#session-information">Session Information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
