<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SeqPlots R workflow • seqplots</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">seqplots</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStart.html">SeqPlots R workflow</a>
    </li>
    <li>
      <a href="../articles/SeqPlotsGUI.html">SeqPlots R workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/przemol/seqplots">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>SeqPlots R workflow</h1>
                        <h4 class="author">Przemyslaw Setmpor</h4>
            
            <h4 class="date">02 August 2017</h4>
          </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>SeqPlots is a tool for plotting average track signals (e.g. read coverage) and sequence motif densities over user specified genomic features. The data can be visualized in linear plots with error estimates or as series of heatmaps that can be sorted and clustered. The software can be run locally on a desktop or deployed on a server and allows easy data sharing. SeqPlots pre-calculates and stores binary result matrices, allowing rapid plot generation. Plots can also be run in batch.</p>
    </div>
    
<div class="contents">
<div id="adding-and-managing-files" class="section level1">
<h1 class="hasAnchor">
<a href="#adding-and-managing-files" class="anchor"></a>Adding and managing files</h1>
<div id="supported-file-formats" class="section level2">
<h2 class="hasAnchor">
<a href="#supported-file-formats" class="anchor"></a>Supported file formats</h2>
<p>Tracks:</p>
<ul>
<li>BigWig (.bw) - <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format6.1" class="uri">http://genome.ucsc.edu/FAQ/FAQformat.html#format6.1</a>
</li>
<li>Wiggle (.wig) - <a href="http://genome.ucsc.edu/goldenPath/help/wiggle.html" class="uri">http://genome.ucsc.edu/goldenPath/help/wiggle.html</a>
</li>
<li>BedGraph (with .wig extension) - <a href="http://genome.ucsc.edu/goldenPath/help/bedgraph.html" class="uri">http://genome.ucsc.edu/goldenPath/help/bedgraph.html</a>
</li>
</ul>
<p>Features:</p>
<ul>
<li>BED - <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1" class="uri">http://genome.ucsc.edu/FAQ/FAQformat.html#format1</a>
</li>
<li>GFF - <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format3" class="uri">http://genome.ucsc.edu/FAQ/FAQformat.html#format3</a>
</li>
<li>GTF (with .gff extension) - <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format4" class="uri">http://genome.ucsc.edu/FAQ/FAQformat.html#format4</a>
</li>
</ul>
<p>Files must be formatted according to UCSC guidelines. All widely used chromosome names conventions are accepted, e.g. for human files either ‘chr1’ or ‘1’ can be used, however these conventions should not be mixed within single files.</p>
</div>
<div id="adding-files" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-files" class="anchor"></a>Adding files</h2>
<p>Press the <code>Add files</code> button to bring up the <strong>file upload panel</strong>.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/05_01.png" alt="File upload panel"><p class="caption">File upload panel</p>
</div>
</blockquote>
<p>You can drag and drop files here or press the <code>Add files...</code> button to open a file selection menu. Before starting the upload the following mandatory information must be provided about each file:</p>
<ul>
<li>User ID</li>
<li>Reference genome - drop-down menu containing reference genome package currently installed in R</li>
</ul>
<p>Comments are optional.</p>
<p>The contents of the text field can be copied to all files by clicking the icon at the left of the field. The default values can be set using <code>Set defaults...</code> button. Default values are stored using the browser cookies, and the settings will be remembered across different sessions as long as the same web browser is used. File extensions that are not supported will raise an error.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/05_02.png" alt="File upload panel with 4 files selected"><p class="caption">File upload panel with 4 files selected</p>
</div>
</blockquote>
<p>Individual files can be uploaded by pressing ‘start’ next to the file name or all files can be uploaded at once by pressing the <code>Start upload</code> button at the top of <strong>file upload panel</strong>.</p>
<p>During the upload process a progress bar is displayed. After upload SeqPlots gives a message that upload was successful or or gives an error message. Common errors are misformatted file formats or chromosome names do not matched the reference genome. For more information please refer to <em>Error explained</em> chapter.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/05_03.png" alt="A feedback on successfully upload files"><p class="caption">A feedback on successfully upload files</p>
</div>
</blockquote>
<p>To dismiss the upload window, click on <code>X</code> or outside the window.</p>
</div>
<div id="downloading-and-removing-files" class="section level2">
<h2 class="hasAnchor">
<a href="#downloading-and-removing-files" class="anchor"></a>Downloading and removing files</h2>
<p>Clicking the <code>New plot set</code> button brings up the <strong>file collection window</strong>. The primary function of this window is to choose signal tracks and feature files to use for calculating the plots. However, it also provides basic file management capabilities. Information on files can be reviewed and files can be downloaded or deleted. Fields can be searched, filtered and sorted by any column. The red <code>x</code> button on the right site of file table removes a single file from the collection, while <code>Remove selected files</code> button will erase all selected files.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/06/06_01.png" alt="The file collection window"><p class="caption">The file collection window</p>
</div>
</blockquote>
</div>
</div>
<div id="running-the-plot-set-jobs" class="section level1">
<h1 class="hasAnchor">
<a href="#running-the-plot-set-jobs" class="anchor"></a>Running the plot-set jobs</h1>
<p>Pressing the <code>New plot set</code> button brings up the <strong>file collection window</strong> from which you can choose signal tracks and feature files to calculate average plots and heat maps. If you wish to upload more files please refer to <a href="Adding%20and%20managing%20files">adding new files documentation</a>. The <strong>file collection window</strong> has three tabs:</p>
<ul>
<li>
<code>Tracks</code> - signal files, i.e., Wiggle, BigWiggle and BedGraph files.</li>
<li>
<code>Features</code> - genomic feature files, i.e., BED, GFF and GTF files</li>
<li>
<code>Sequence features</code> - input any motif of interest that you want to plot.</li>
</ul>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/06/06_01.png" alt="The file collection modal"><p class="caption">The file collection modal</p>
</div>
</blockquote>
<div id="selecting-files" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-files" class="anchor"></a>Selecting files</h2>
<p>The <code>Tracks</code> and <code>Features</code> tabs displays information about the files and allows you to filter and sort by any column. The “Search:” dialog allows you to find any keyword in any field, while dropdowns below the file grid allow for more advanced filtering on specific columns.</p>
<p>Select files by clicking on the file name or any other part of the row beside <code>Show comment</code> and <code>Download</code> or <code>Remove</code> buttons. Chosen files are highlighted in light blue. Clicking the file name again will cancel the selection. At least one signal track or motif and one feature file must be selected before starting the calculation.</p>
</div>
<div id="setting-up-plot-options" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-up-plot-options" class="anchor"></a>Setting up plot options</h2>
<p>Options controlling the plot settings is found below the file selection window:</p>
<ol style="list-style-type: decimal">
<li>
<strong><code>Bin track @ [bp]:</code></strong> - this numeric input determines the resolution of data acquisition; the default value 10 means that 10bp intervals within the plotting range will be summarized by calculating the mean. Higher values increases the speed of calculation, but decreases resolution. See the <a href="Terms">explanations</a>.</li>
<li>
<strong><code>Choose the plot type</code></strong> - there are three options:
<ul>
<li>
<em><code>Point Features</code></em> - anchor plot on the start of a feature. By default, plot will be directional if strand information is present (i.e, use start position and plot on positive strand for + strand features and use end position and plot on negative strand for minus strand features). If strand information is not present in the feature file (or if the “ignore strand” option is chosen), plot will use start position of feature and be plotted on the positive strand (see <a href="Terms">explanations</a>). User chooses length of upstream and downstream sequence to plot.</li>
<li>
<em><code>Midpoint Features</code></em> - similar to point feature, but plot is centered on the midpoint of the feature.</li>
<li>
<em><code>Endpoint Features</code></em> - similar to point feature, but plot is centered on the end of the feature. Strand information is used by default to determine the end side.</li>
<li>
<em><code>Anchored Features</code></em> - features are anchored at start and stop positions and given pseudo-length chosen by the user. Additionally, the user chooses the length of sequence upstream of the start and downstream of the end to plot.</li>
</ul>
</li>
<li>
<strong><code>Ignore strand</code></strong> - the directionality (strand) will be ignored all features plotted on the positive strand.</li>
<li>
<strong><code>Ignore zeros</code></strong> - signal values of 0 in the track will be be excluded from calculations</li>
<li>
<strong><code>Calculate heatmap</code></strong> - selecting this generates and saves a heat map matrix. Select if you wish to generate heatmap; uncheck if you only wish to generate average plots, as this will speed calculations.</li>
<li>
<strong><code>Plotting distances in [bp]</code></strong> - the distances in to be plotted:
<ul>
<li>
<em><code>Upstream</code></em> - the plotting distance in base pairs upstream to the feature</li>
<li>
<em><code>Anchored</code></em> - the pseudo-length, to which the features will be extended or shrunk using linear approximation (only for anchored plots)</li>
<li>
<em><code>Downstream</code></em> - the plotting distance in base pairs downstream to the feature</li>
</ul>
</li>
</ol>
</div>
<div id="plotting-sequence-motif-density" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-sequence-motif-density" class="anchor"></a>Plotting sequence motif density</h2>
<p>The <code>Sequence features</code> tab allows you to calculate and plot the density of any user-defined motif around the chosen genomic feature using the reference sequence package. Motif plots can be mixed with track files’ signal plots. The following options can be set:</p>
<ol style="list-style-type: decimal">
<li>
<strong><code>DNA motif</code></strong> - the DNA motif</li>
<li>
<strong><code>Sliding window size in base pairs [bp]</code></strong> - the size of the sliding window for motif calculation. The value (number of matching motifs within the window) is reported in the middle of the window, e.g. if window is set to 200bp, DNA motif is “GC” and there are 8 CpGs in first 200 bp of the chromosome the value 8 will be reported at 100th bp.</li>
<li>
<strong><code>Display name</code></strong> - The name of the motif that will be shown in key and heatmap labels. Leave blank to use <code>DNA motif</code> value.</li>
<li>
<strong><code>Plot heatmap or error estimates</code></strong> - this checkbox determines if heatmap matrix and error estimates should be calculated. If unchecked much faster algorithm will be used for motif density calculation, but only the average plot without the error estimates will be available.</li>
<li>
<strong><code>Match reverse complement as well</code></strong> - select if reverse complement motif should be reported as well. For example the TATA motif will report both TATA and ATAT with this option selected.</li>
</ol>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/06/06_02.png" alt="Sequence motifs selection tab"><p class="caption">Sequence motifs selection tab</p>
</div>
</blockquote>
<p>Clicking <code>Add</code> button adds the motif to plot set, while <code>Reset All</code> clears the motif selection. On the right side of the motif setting panel gives a list summary of included motifs.</p>
</div>
<div id="starting-the-plot-set-calculation" class="section level2">
<h2 class="hasAnchor">
<a href="#starting-the-plot-set-calculation" class="anchor"></a>Starting the plot set calculation</h2>
<p>The options are executed by pressing the <code>Run calculation</code> button. This dismisses the <strong>file collection window</strong> and brings up the calculation dialog, which shows the progress. On Linux and Mac OS X (systems supporting fork based parallelization) the calculation can be stopped using the <code>Cancel</code> button - this will bring back all settings in <strong>file collection window</strong>.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/06/06_03.png" alt="The calculation progress dialog"><p class="caption">The calculation progress dialog</p>
</div>
</blockquote>
<p>After successful execution the <strong>plot array</strong> and <strong>plot preview panel</strong> will appear. In case of error an informative error pop-up will explain the problem. Please refer to the error section for further information.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/06/06_04.png" alt="The plot array"><p class="caption">The plot array</p>
</div>
</blockquote>
</div>
</div>
<div id="plotting" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting" class="anchor"></a>Plotting</h1>
<blockquote>
<p>This section focuses on average (line) plots and options common between these and heatmaps. For heatmap options please refer to heatmap documentation.</p>
</blockquote>
<div id="previewing-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#previewing-plot" class="anchor"></a>Previewing plot</h2>
<p>After calculating or loading a plot set, a <strong>plot array</strong> of checkboxes is displayed to select the desired pairs of features and tracks/motifs. Clicking on the column name (tracks/motifs) or row name (features) selects/deselects the whole column or row. Clicking on top-left most cell of <strong>plot array</strong> toggles the selection of whole array.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/01_QuickStart/01_QuickStart_08.png" alt="Plot preview plus Line plot, Heatmap and refresh buttons"><p class="caption">Plot preview plus <code>Line plot</code>, <code>Heatmap</code> and <code>refresh</code> buttons</p>
</div>
</blockquote>
<p>If at least one pair on <strong>plot array</strong> is selected pressing the <code>Line plot</code> button produces an average plot preview and the <code>Heatmap</code> button produces a heatmap preview. Alternatively, pressing the [RETURN] key will also produce the new selection and options. These operations are done automatically in reactive mode (see <strong>Advanced options</strong> chapter). Plots can be downloaded as PDF files using the Line plot and Heatmap buttons next to Download (at the top of the panel).</p>
<p>Below the plotting buttons are options for labeling plots and setting axes. On application start the first panel responsible for bringing file upload, management and plot set calculation modals is active. The further three panels hold common plot settings.</p>
</div>
<div id="titles-and-axis-panel" class="section level2">
<h2 class="hasAnchor">
<a href="#titles-and-axis-panel" class="anchor"></a>Titles and axis panel</h2>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/07_03.png" alt="The view on titles and axis panel"><p class="caption">The view on titles and axis panel</p>
</div>
</blockquote>
<p>This panel groups settings influencing the plot main title, axis labels, various font sizes plus vertical and horizontal plot limits.</p>
<ul>
<li>
<code>Title</code> - The main title of the plot, shown in top-center part of the figure; default empty</li>
<li>
<code>X-axis label</code> - Label shown below horizontal axis; default empty</li>
<li>
<code>Y-axis label</code> - Label shown below vertical axis; default empty</li>
<li>
<code>Title font size</code> - Font size of the title in points (point = ~1/72 an inch for standard A4 output); default 20 points</li>
<li>
<code>Labels font size</code> - Font size of axis labels in points; default 16 points</li>
<li>
<code>Axis font size</code> - Controls axis ticks font size, that is size of the numbers indicating position in base pairs on X-axis and means signal value on X-axis; default 14 points</li>
<li>
<code>Set X-axis limits</code> - Set hard plotting limits for X-axis; default values are whole range chosen during plot set calculation</li>
<li>
<code>Set Y-axis limits</code> - Set hard plotting limits for Y-axis; default values are a range between lowest and highest mean signal extended by error estimate</li>
</ul>
</div>
<div id="guide-lines-and-data-scaling" class="section level2">
<h2 class="hasAnchor">
<a href="#guide-lines-and-data-scaling" class="anchor"></a>Guide lines and data scaling</h2>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/07_04.png" alt="The view on guide lines and data scaling"><p class="caption">The view on guide lines and data scaling</p>
</div>
</blockquote>
<p>Controls in this panel controls the display of guide lines and error estimates, and allows to log scale the signal prior to plotting.</p>
<ul>
<li>
<code>Transform signal</code> - if set to <em><code>Log2 transform</code></em> performs log2 transformation of the signal prior to plotting; default setting is <em><code>Do not transform</code></em>
</li>
<li>
<code>Show vertical guide line</code> - show the vertical line at point 0 - beginning of the feature or midpoint and end of the pseudo-length scaled features (only for anchored plots); turn on by default</li>
<li>
<code>Show horizontal guide line</code> - show the horizontal line at user determined height; turn off by default</li>
<li>
<code>Show error estimates</code> - show error standard error and 95% confidence interval as fields, if turned off only the line representing the mean signal is shown; turn on by default</li>
</ul>
</div>
<div id="keys-labels-and-colors-panel" class="section level2">
<h2 class="hasAnchor">
<a href="#keys-labels-and-colors-panel" class="anchor"></a>Keys, labels and colors panel</h2>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/07_05_06.png" alt="The view on keys, labels and colors panel (left). Color picker, label text input and Priority/Order checkboxes reviled on plot set grid (right)."><p class="caption">The view on keys, labels and colors panel (left). Color picker, label text input and Priority/Order checkboxes reviled on plot set grid (right).</p>
</div>
</blockquote>
<p>This panel groups two types of controls. <code>Colors</code>, <code>Label</code> and <code>Priority/Order</code> are a checkboxes revealing further controls on <strong>plot set grid</strong>, specific for a feature-track pair or sub-heatmap. <code>Show plot key</code>, <code>Show error estimate key</code> and <code>Legend font size</code> re global controls specific for average plots. Inputs on <strong>plot set grid</strong> do not have specific labels, but the tooltip explaining their meaning is shown on mouse cursor hover.</p>
<ul>
<li>
<code>Colors</code> - checkboxes revealing a color picker on <strong>plot set grid</strong>. This input allows to control the colors of specific feature-track pair average plots or sub-heatmaps. In browser supporting the color picker ‘e.g Chrome’ the system dialog will show up. In other browsers (e.g. Firefox) the javaScript color picker will be initialized.</li>
<li>
<code>Label</code> - checkboxes revealing a label text input <strong>plot set grid</strong>. This controls the names shown on the <strong>key</strong> with average plots or the heatmap top labels.</li>
<li>
<code>Priority/Order</code> - checkboxes revealing numeric input on <strong>plot set grid</strong>. These number determine the order of average plots and hetamaps. Feature-track pair with the highest priority will be listed on the top of <strong>key</strong> for average plots and left-most for heatmaps.</li>
<li>
<code>Show plot key</code> - shows the key giving the color to feature-track pair label mapping. If turned on the additional drop-down allows to choose the position on the plot, top-right by default</li>
<li>
<code>Show error estimate key</code> - shows the key gexplaining the meaning of error fields. If turnedon the additional drop-down allows to choose the position on the plot, top-left by default</li>
<li>
<code>Legend font size</code> - set the size of font used to plot the keys; 12 default Heatmaps ========</li>
</ul>
<p>Heatmaps are often more informative than average plots. If there is variability in signal along individual instances of a given genomic feature (e.g., because there are different biological classes), an average plot might not represent the behavior of any individual feature and could even give a misleading picture. SeqPlots plots track-feature pairs as sub-heatmaps horizontally aligned on single figure. All sub-heatmaps must have the same number of data rows, hence in single plot mode simultaneous plotting is possible only on single features or feature files containing exact same number of rows. The heatmaps can be sorted and clustered by k-means, hierarchical clustering or super self organising maps (SupreSOM).</p>
</div>
<div id="heatmap-setup-tab" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmap-setup-tab" class="anchor"></a>Heatmap setup tab</h2>
<p>This tab has heatmap specific options for data processing and display.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/08_02.png" alt="The view on Heatmap setup tab (left). Color picker, Label text input, Priority/Order checkboxes, Choose individual heatmaps for sorting/clustering control and Set individual color key limits numric inputs reviled on plot set grid (right)."><p class="caption">The view on Heatmap setup tab (left). Color picker, Label text input, Priority/Order checkboxes, Choose individual heatmaps for sorting/clustering control and Set individual color key limits numric inputs reviled on plot set grid (right).</p>
</div>
</blockquote>
<ul>
<li><p><strong><code>Sort heatmap rows by mean signal</code></strong> - sorts the heatmap rows based on the mean value of each row across included sub-heatmps. Can be set to increasing or decreasing order. Turned off by default.</p></li>
<li><p><strong><code>Clustering algorithm</code></strong> - choose clustering algorithm (k-means, hierarchical or SupreSOM). If clustering is not desired, choose <em><code>do not cluster</code></em>, which uses the feature file in the uploaded order . K-means by default.</p></li>
<li><p><strong><code>Make cluster calculation repeatable</code></strong> - enforces, that clustering with non-deterministic algorithms, like k-means or SupreSOM will generate the same results as most recently plotted heatmap. This is achieved by re-using R random number generator seed.</p></li>
<li><p><strong><code>Plot selected cluster</code></strong> - this option is available only if <code>Make cluster calculation repeatable</code> is turned on. Allows to select one of the clusters and zoom it to whole plot height. Plot all clusters by default.</p></li>
<li><p><strong><code>Choose individual heatmaps for sorting/clustering</code></strong> - this checkbox brings up a new control panel on the <strong>plot set grid</strong> to determine if a given sub-heatmap should be included in plotting and/or clustering. The excluded sub-plots will be plotted in the order of the other sub-heatmaps, but their values will not influence the clustering/sorting. By default all sub-heatmaps are included.</p></li>
<li>
<p><strong><code>Heatmaps have individual color keys</code></strong> - by default all sub-heatmap have their own color keys. This option determines if each sub-heatmap should have a separate color key (plotted below the heatmap) or a single, common key should be calculated for all sub-plots (plotted rightmost). The example below show the difference between separate (left) and common (right) color keys:</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/08_04.png" alt="Heatmaps have individual color keys usage example: separate (left) and common (right) color keys"><p class="caption"><code>Heatmaps have individual color keys</code> usage example: separate (left) and common (right) color keys</p>
</div>
</blockquote>
</li>
<li>
<p><strong><code>Set default color key limits</code></strong> - this option determines the limits in mapping the numerical values to the colors. The range of colors generated is dependent on these options. Values lower or higher than the given limits will be plotted in the limit value color. If this checkbox is not selected, limits are auto-generated using <strong><code>Color key scaling</code></strong> parameter. If this option it turned off two numerical fields, <strong><code>min</code></strong> and <strong><code>max</code></strong>, are shown to manually set the limits.</p>
<ul>
<li>
<strong><code>Color key scaling</code></strong> - this slider influences how color key limits are generated. For example, 0.01 (default value) calculates limits using data ranging from 1-99 percentile of available data points. 0.1 uses data ranging from 10-90 percentile. The general formula for limit is: [quantile(data, <code>Color key scaling</code>); quantile(data, 1-<code>Color key scaling</code>)]</li>
<li>
<strong><code>min</code></strong> and <strong><code>max</code></strong> numeric inputs - enter values to manually specify color key limits as numeric values.</li>
</ul>
</li>
<li><p><strong><code>Set individual color key limits</code></strong> - this option is similar to manual set up of color key limits, but this allows one to specify different values for individual sub-heatmaps. When this checkbox is selected <strong><code>min</code></strong> and <strong><code>max</code></strong> numeric input menu is shown on the <strong>plot set grid</strong></p></li>
<li>
<p><strong><code>Set colorspace</code></strong> - This input allows to use color palettes from RColorBrewer package, which would replace default color palette for heatmaps. By selecting <strong><code>Reverse</code></strong> checkbox the reversed color palette will be used. Here are available color pallets (click <a href="http://przemol.github.io/plotsHg19/ColorMapTests">here</a> to see example heatmap plotted with different color palettes):</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/palettes.png" alt="Color palettes from RColorBrewer"><p class="caption">Color palettes from RColorBrewer</p>
</div>
<p>When the <code>Custom</code> option is selected three-color pickers are shown to setup custom color mappings for heatmaps. The following example below shows standard jet colors (left), default blue color mapping after selecting the checkbox (middle) and custom color selection (right):</p>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/08_05.png" alt="Set default colorspace usage example: standard jet colors (left), default blue color mapping after selecting the checkbox (middle) and custom color selection (right)"><p class="caption"><code>Set default colorspace</code> usage example: standard jet colors (left), default blue color mapping after selecting the checkbox (middle) and custom color selection (right)</p>
</div>
</li>
</ul>
</div>
<div id="other-options-controlling-heatmap-appearance" class="section level2">
<h2 class="hasAnchor">
<a href="#other-options-controlling-heatmap-appearance" class="anchor"></a>Other options controlling heatmap appearance</h2>
<p>The heatmap output shares many display options from other tabs. Here we provide a list of these inputs, please refer to <a href="Viewing%20and%20manipulating%20plots">“Viewing and manipulating plots”</a> for further reference.</p>
<ul>
<li>
<strong>Titles and axis panel</strong>
<ul>
<li>
<code>X-axis label</code> - Label shown below horizontal axis, drawn separately for each sub-heatmap; default empty</li>
<li>
<code>Y-axis label</code> - Label shown next to vertical axis, drawn separately for each sub-heatmap; default empty</li>
<li>
<code>Labels font size</code> - Font size for axis labels and main labels of sub-heatmaps; default 16 points</li>
<li>
<code>Axis font size</code> - Controls axis ticks font size; default 14 points</li>
<li>
<code>Set X-axis limits</code> - Set hard plotting limits for X-axis; default values are whole range chosen during plot set calculation</li>
</ul>
</li>
<li>
<strong>Guide lines and data scaling panel</strong>
<ul>
<li>
<code>Transform signal</code> - if set to <em><code>Log2 transform</code></em> performs log2 transformation of the signal prior to plotting; default setting is <em><code>Do not transform</code></em>
</li>
<li>
<code>Show vertical guide line</code> - show the vertical line at point 0 - beginning of the feature or midpoint and end of the pseudo-length scaled features (only for anchored plots); turn on by default</li>
</ul>
</li>
<li>
<strong>Keys, labels and colors panel</strong>
<ul>
<li>
<code>Colors</code> - for hetmaps this input allows to control the color mapping of specific sub-heatmaps. The map starts with white (for low color key limit) and finishes with selected color (for high color key limit).</li>
<li>
<code>Label</code> - allows to set up custom sub-heatmap top labels</li>
<li>
<code>Priority/Order</code> - Use this to place heatmaps in your desired order. The feature-track pairs with the highest priority will be plotted as left-most sub-heatmaps.</li>
<li>
<code>Legend font size</code> - control the font size of common color key, inactive if heatmaps have individual color keys; 12 default</li>
</ul>
</li>
</ul>
</div>
</div>
<div id="getting-pdfs-and-cluster-info" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-pdfs-and-cluster-info" class="anchor"></a>Getting PDFs and cluster info</h1>
<p>Plots can be downladed as PDFs by clicking <code>Line plot</code> or <code>Heatmap</code> buttons in the “Download:” section of the <strong>tool panel</strong> (above the plot preview).</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/09_01.png" alt='Download:" section of tool panel with Line plot and Heatmap buttons'><p class="caption">Download:" section of tool panel with <code>Line plot</code> and <code>Heatmap</code> buttons</p>
</div>
</blockquote>
<p>The small buttons next to <code>Line plot</code> and <code>Heatmap</code> produce additional output files:</p>
<ul>
<li>the <code>i</code> button next to <code>Line plot</code> downloads the PDF containing average plot keys</li>
<li>
<code>cluster diagram</code> button next to <code>Heatmap</code> downloads a cluster report giving cluster assignments and sorting order for each feature as a comma separated value (CSV) spreadsheet.</li>
</ul>
<p>The cluster report contains following columns:</p>
<ul>
<li>
<code>chromosome</code> - the name of chromosome, contig or scaffold</li>
<li>
<code>start</code> - start of the feature (1 based chromosomal coordinate)</li>
<li>
<code>end</code> - end of the feature (1 based chromosomal coordinate)<br>
</li>
<li>
<code>width</code> - width of the feature in base pairs</li>
<li>
<code>strand</code> - strand of the feature</li>
<li>
<code>metadata_...</code> - annotation columns present in the original GFF/BED e.g. gene name, score, group</li>
<li>
<code>originalOrder</code> - number of feature (row) in GFF/BED, can be used to restore original order after sorting on cluster ID</li>
<li>
<code>ClusterID</code> - the numeric ID of the cluster. The topmost cluster on the heatmap is annotated with 1, and the bottom cluster with k, where k equals to number of clusters selected, exported only if clustering is enabled</li>
<li>
<code>SortingOrder</code> - the order imposed on heatmap by sorting by mean row(s) values, exported only if sorting is enabled</li>
<li>
<code>FinalOrder</code> - the final order of heatmap’s rows, this can be influenced by sorting and clustering; 1 indicates topmost row</li>
</ul>
<p>Sample report: </p>
<pre><code>chromosome  start   end     width   strand  metadata_group  originalOrder   ClusterID   SortingOrder    FinalOrder
chrI        9065087 9070286 5200    +       g1              1               1           3               3
chrI        5171285 5175522 4238    -       g1              2               3           50              43
chrI        9616508 9618109 1602    -       g1              3               3           13              43
chrI        3608395 3611844 3450    +       g1              4               3           11              12</code></pre>
<p></p>
<p>Table view: </p>
<table style="width:100%;" class="table">
<colgroup>
<col width="10%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="6%">
<col width="13%">
<col width="12%">
<col width="8%">
<col width="11%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th>chromosome</th>
<th>start</th>
<th>end</th>
<th>width</th>
<th>strand</th>
<th>metadata_group</th>
<th>originalOrder</th>
<th>ClusterID</th>
<th>SortingOrder</th>
<th>FinalOrder</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>chrI</td>
<td>9065087</td>
<td>9070286</td>
<td>5200</td>
<td>+</td>
<td>g1</td>
<td>1</td>
<td>1</td>
<td>3</td>
<td>3</td>
</tr>
<tr class="even">
<td>chrI</td>
<td>5171285</td>
<td>5175522</td>
<td>4238</td>
<td>-</td>
<td>g1</td>
<td>2</td>
<td>3</td>
<td>50</td>
<td>43</td>
</tr>
<tr class="odd">
<td>chrI</td>
<td>9616508</td>
<td>9618109</td>
<td>1602</td>
<td>-</td>
<td>g1</td>
<td>3</td>
<td>3</td>
<td>13</td>
<td>43</td>
</tr>
<tr class="even">
<td>chrI</td>
<td>3608395</td>
<td>3611844</td>
<td>3450</td>
<td>+</td>
<td>g1</td>
<td>4</td>
<td>3</td>
<td>11</td>
<td>12</td>
</tr>
</tbody>
</table>
<p></p>
<div id="pdf-output-size" class="section level2">
<h2 class="hasAnchor">
<a href="#pdf-output-size" class="anchor"></a>PDF output size</h2>
<p>The last tab (<code>Batch operation and setup</code>) on the <strong>tool panel</strong> includes batch operations and various other settings including PDF output size. By default the output PDF will be A4 landscape. This can be changed using the drop-down list to following settings:</p>
<ul>
<li>
<code>user defined</code> - this option reveals two numeric inputs that allows to set output PDF width and height. The values must be given in inches.</li>
<li>
<code>Legal rotated</code> - US Legal landscape: 14" by 8.5"</li>
<li>
<code>A4</code> - A4 portrait: - 8.27" × 11.69"</li>
<li>
<code>Letter</code> - US Letter portrait: 8.5" × 11"</li>
<li>
<code>Legal</code> - US Legal portrait: 8.5" × 14"</li>
<li>
<code>Executive</code> - a.k.a Monarch paper: 7.25 × 10.5"</li>
</ul>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/09_02.png" alt="The view on top part of batch operation and setup panel"><p class="caption">The view on top part of batch operation and setup panel</p>
</div>
</blockquote>
</div>
</div>
<div id="batch-operations" class="section level1">
<h1 class="hasAnchor">
<a href="#batch-operations" class="anchor"></a>Batch operations</h1>
<p>Controls to plot multiple plots at once are located on the <code>Batch operation and setup</code> tab, just below PDF paper options. It is possible to output the plots to multipage PDF, plot an array of plots on a single page (for average plots) or mix these options together.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/09_03.png" alt="The view on bottom part of batch operation and setup panel"><p class="caption">The view on bottom part of batch operation and setup panel</p>
</div>
</blockquote>
<p>The first drop-down controls the type of the plot - either average or heatmap. The second drop down determines the strategy to traverse the <strong>plot grid</strong>. The options include:</p>
<ul>
<li>
<code>single</code> - every single feature-track pair will be plotted on separate plot</li>
<li>
<code>rows</code> - the <strong>plot grid</strong> will be traversed by rows, which means one plot that contains all tracks per feature will be prepared</li>
<li>
<code>columns</code> - the <strong>plot grid</strong> will be traversed by columns, which means one plot that contains all features per tracks will be prepared</li>
</ul>
<p>The <code>multi plot grid</code> option controls how many plots will be placed on each page of the PDF output, e.g. 1x1 means one plot per one page, while 3x4 means 3 columns and 4 rows of plots. If number of plots exceeds the number of slots on page the new page will be added to the PDF.</p>
<p><code>Filter names</code> will apply a filter to plot titles, which are based on on uploaded file names. For example, if you uploaded 100 files starting with a prefix of “my_experiment_”, you can remove this fragment from each plot title and/or heatmap caption by putting this string in <code>Filter names</code>.</p>
<p>Finally, pressing <code>Get PDF</code> produces the final output file. Please see example below:</p>
<blockquote>
<p><img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/09_04.png" alt="Batch plot usage example - multiple average plots aranged in 6x2 plot grid"> Saving and loading plotsets ============================</p>
</blockquote>
<p>If desired, SeqPlots will save plot sets as binary R files, allowing you to quickly load the pre-calculated set for replotting. Saved plot sets can also be shared with other SeqPlots users.</p>
<div id="load-or-save-plotset" class="section level2">
<h2 class="hasAnchor">
<a href="#load-or-save-plotset" class="anchor"></a>Load or save plotset</h2>
<p>Controls available on the “Load or save plotset” panel:</p>
<ul>
<li>
<strong><code>Load saved plot set</code></strong> - drop-down list to select a plotset. Once the Rdata binary file is selected the <strong>plot grid</strong> will be displayed. Selecting the file reveals two additional buttons:
<ul>
<li>
<strong><code>Remove dataset</code></strong> - this button deletes the selected saved plot set from user data.</li>
<li>
<strong><code>Download plotset</code></strong> - this button saves a copy of the plotset in selected location.</li>
</ul>
</li>
<li>
<strong><code>Save current plot set</code></strong> - Enter desired name and press the <code>Save</code> button (appears after input of name). It is also possible to save a copy of loaded plot sets. The plot set binary files can be renamed simply by loading them, saving a copy and deleting original source file.</li>
</ul>
<p>All saved dataset can be found in <code>data location</code>/publicFiles. Any SeqPlots Rdata binaries put in the folder will become available for loading in <strong><code>Load saved plot set</code></strong> control.</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/10_00.png" alt="The view on the Load or save plotset panel"><p class="caption">The view on the “Load or save plotset” panel</p>
</div>
</blockquote>
</div>
<div id="plot-set-files-structure" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-set-files-structure" class="anchor"></a>Plot set files structure</h2>
<p>The plot sets files can be also directly loaded in R. This allows further processing and customization of the plots. Data structure is a nested list, which elements be accessed by <code>[[</code> R operator. The nesting goes as follow:</p>
<ul>
<li>
<strong><code>feature</code></strong> - R list
<ul>
<li>
<strong><code>track</code></strong> - R list
<ul>
<li>
<code>means</code> - numeric vector giving mean signal value for each (binned) genomic position</li>
<li>
<code>stderror</code> - numeric vector giving standard error for each (binned) genomic position</li>
<li>
<code>conint</code> - numeric vector giving 95% confidence interval for each (binned) genomic position</li>
<li>
<code>all_ind</code> - numeric vector giving the genomic position in base pairs</li>
<li>
<code>e</code> - character string giveing numeric vector giving the indicates of anchored distance, NULL for point features plots</li>
<li>
<code>desc</code> - auto generated title of the plot</li>
<li>
<code>heatmap</code> - numeric matrix, (binned) signal values for each genomic position (columns) and each feature (rows)</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>The example structure:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">List of <span class="dv">2</span>
 $<span class="st"> </span>HTZ1_Differential_genes_TOP100_v2.gff:List of <span class="dv">2</span>
  ..$<span class="st"> </span>HTZ1_JA00001_IL1andIL2_F_N2_L3_NORM_linear_1bp_IL010andIL009_averaged.bw    :List of <span class="dv">7</span>
  .. ..$<span class="st"> </span>means   :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">2.52</span> <span class="fl">2.52</span> <span class="fl">2.52</span> <span class="fl">2.53</span> <span class="fl">2.54</span> ...
  .. ..$<span class="st"> </span>stderror:<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.114</span> <span class="fl">0.112</span> <span class="fl">0.111</span> <span class="fl">0.11</span> <span class="fl">0.109</span> ...
  .. ..$<span class="st"> </span>conint  :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.226</span> <span class="fl">0.223</span> <span class="fl">0.221</span> <span class="fl">0.218</span> <span class="fl">0.217</span> ...
  .. ..$<span class="st"> </span>all_ind :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] -<span class="dv">1000</span> -<span class="dv">995</span> -<span class="dv">990</span> -<span class="dv">985</span> -<span class="dv">980</span> -<span class="dv">975</span> -<span class="dv">970</span> -<span class="dv">965</span> -<span class="dv">960</span> -<span class="dv">955</span> ...
  .. ..$<span class="st"> </span>e       :<span class="st"> </span><span class="ot">NULL</span>
  .. ..$<span class="st"> </span>desc    :<span class="st"> </span>chr <span class="st">"HTZ1_JA00001_IL1andIL2...</span><span class="ch">\n</span><span class="st">@HTZ1_Differential_genes_TOP100_v2"</span>
  .. ..$<span class="st"> </span>heatmap :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">100</span>, <span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">2.36</span> <span class="fl">5.25</span> <span class="fl">2.2</span> <span class="fl">3.48</span> <span class="fl">4.32</span> ...
  ..$<span class="st"> </span>HTZ1_JA00001_IL3andIIL5_F_lin35_L3_NORM_linear_1bp_IL008andIL011_averaged.bw:List of <span class="dv">7</span>
  .. ..$<span class="st"> </span>means   :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">2.36</span> <span class="fl">2.35</span> <span class="fl">2.35</span> <span class="fl">2.36</span> <span class="fl">2.38</span> ...
  .. ..$<span class="st"> </span>stderror:<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.126</span> <span class="fl">0.125</span> <span class="fl">0.125</span> <span class="fl">0.126</span> <span class="fl">0.125</span> ...
  .. ..$<span class="st"> </span>conint  :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.249</span> <span class="fl">0.249</span> <span class="fl">0.247</span> <span class="fl">0.251</span> <span class="fl">0.249</span> ...
  .. ..$<span class="st"> </span>all_ind :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] -<span class="dv">1000</span> -<span class="dv">995</span> -<span class="dv">990</span> -<span class="dv">985</span> -<span class="dv">980</span> -<span class="dv">975</span> -<span class="dv">970</span> -<span class="dv">965</span> -<span class="dv">960</span> -<span class="dv">955</span> ...
  .. ..$<span class="st"> </span>e       :<span class="st"> </span><span class="ot">NULL</span>
  .. ..$<span class="st"> </span>desc    :<span class="st"> </span>chr <span class="st">"HTZ1_JA00001_IL3andIIL5...</span><span class="ch">\n</span><span class="st">@HTZ1_Differential_genes_TOP100_v2"</span>
  .. ..$<span class="st"> </span>heatmap :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">100</span>, <span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">2.61</span> <span class="fl">3.17</span> <span class="fl">1.42</span> <span class="fl">2.46</span> <span class="fl">4.26</span> ...
 $<span class="st"> </span>HTZ1_Differential_genes_BOTTOM100.gff:List of <span class="dv">2</span>
  ..$<span class="st"> </span>HTZ1_JA00001_IL1andIL2_F_N2_L3_NORM_linear_1bp_IL010andIL009_averaged.bw    :List of <span class="dv">7</span>
  .. ..$<span class="st"> </span>means   :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">1.57</span> <span class="fl">1.57</span> <span class="fl">1.58</span> <span class="fl">1.6</span> <span class="fl">1.62</span> ...
  .. ..$<span class="st"> </span>stderror:<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.0996</span> <span class="fl">0.0985</span> <span class="fl">0.1003</span> <span class="fl">0.1022</span> <span class="fl">0.1018</span> ...
  .. ..$<span class="st"> </span>conint  :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.198</span> <span class="fl">0.195</span> <span class="fl">0.199</span> <span class="fl">0.203</span> <span class="fl">0.202</span> ...
  .. ..$<span class="st"> </span>all_ind :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] -<span class="dv">1000</span> -<span class="dv">995</span> -<span class="dv">990</span> -<span class="dv">985</span> -<span class="dv">980</span> -<span class="dv">975</span> -<span class="dv">970</span> -<span class="dv">965</span> -<span class="dv">960</span> -<span class="dv">955</span> ...
  .. ..$<span class="st"> </span>e       :<span class="st"> </span><span class="ot">NULL</span>
  .. ..$<span class="st"> </span>desc    :<span class="st"> </span>chr <span class="st">"HTZ1_JA00001_IL1andIL2...n@HTZ1_Differential_genes_BOTTOM100"</span>
  .. ..$<span class="st"> </span>heatmap :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">100</span>, <span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">1.64</span> <span class="fl">1.37</span> <span class="fl">1.61</span> <span class="fl">1.77</span> <span class="fl">1.86</span> ...
  ..$<span class="st"> </span>HTZ1_JA00001_IL3andIIL5_F_lin35_L3_NORM_linear_1bp_IL008andIL011_averaged.bw:List of <span class="dv">7</span>
  .. ..$<span class="st"> </span>means   :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">1.94</span> <span class="fl">1.94</span> <span class="fl">1.95</span> <span class="fl">1.96</span> <span class="fl">1.97</span> ...
  .. ..$<span class="st"> </span>stderror:<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.123</span> <span class="fl">0.123</span> <span class="fl">0.124</span> <span class="fl">0.126</span> <span class="fl">0.128</span> ...
  .. ..$<span class="st"> </span>conint  :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">0.244</span> <span class="fl">0.245</span> <span class="fl">0.246</span> <span class="fl">0.251</span> <span class="fl">0.253</span> ...
  .. ..$<span class="st"> </span>all_ind :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">501</span>] -<span class="dv">1000</span> -<span class="dv">995</span> -<span class="dv">990</span> -<span class="dv">985</span> -<span class="dv">980</span> -<span class="dv">975</span> -<span class="dv">970</span> -<span class="dv">965</span> -<span class="dv">960</span> -<span class="dv">955</span> ...
  .. ..$<span class="st"> </span>e       :<span class="st"> </span><span class="ot">NULL</span>
  .. ..$<span class="st"> </span>desc    :<span class="st"> </span>chr <span class="st">"HTZ1_JA00001_IL3andIIL5...</span><span class="ch">\n</span><span class="st">@HTZ1_Differential_genes_BOTTOM100"</span>
  .. ..$<span class="st"> </span>heatmap :<span class="st"> </span>num [<span class="dv">1</span>:<span class="dv">100</span>, <span class="dv">1</span>:<span class="dv">501</span>] <span class="fl">1.61</span> <span class="fl">1.37</span> <span class="fl">1.29</span> <span class="fl">3.04</span> <span class="fl">3.77</span> ...</code></pre></div>
</div>
</div>
<div id="advanced-options" class="section level1">
<h1 class="hasAnchor">
<a href="#advanced-options" class="anchor"></a>Advanced options</h1>
<p>Some additional SeqPlots options are located at very bottom of the <code>Batch operation and setup</code> tab:</p>
<blockquote>
<div class="figure">
<img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/10_01.png" alt="The view on Advanced options section of the batch operation and setup panel"><p class="caption">The view on ‘Advanced options’ section of the batch operation and setup panel</p>
</div>
</blockquote>
<ul>
<li><p><code>Keep 1:1 aspect ratio in batch mode</code> - This option guarantee that the ratio between X- and Y-axis height will be 1, hence the produced plots will be square in batch mode. This prevents stretching the plots while fitting single rows or columns to one page. Turned on by default.</p></li>
<li><p><code>Always keep 1:1 aspect ratio</code> - This checkbox extends the 1:1 aspect ratio option to single plots. Turned off by default.</p></li>
<li><p><code>Reactive plotting</code> - When selected, all plotting operations are executed upon selection and will be visible in preview. <code>Reactive plotting</code> might be useful for exploratory data analysis using plots, but it is not recommended for heatmap plots because speed is decreased. Select/delelect from keyboard by pressing [ctrl/cmd+R]. Turned off by default.</p></li>
<li><p><code>Use multithreading for calculations</code> - This option is available only on desktop instances of SeqPlots under Mac OS X and Linux. While turned off R will not fork the child processes for plotting and plot set calculations. It is useful for debugging, since in single process mode all warning/errors will be directly printed to R console. Also might increase the performance for plotting small average plots. Turned off by default.</p></li>
<li><p><code>Use ggplot2 graphics package for heatmaps</code> - uses GGplot2 (<a href="http://ggplot2.org/" class="uri">http://ggplot2.org/</a>) graphics system to draw the heatmaps. This feature is experimental.</p></li>
</ul>
</div>
<div id="genomes-managment" class="section level1">
<h1 class="hasAnchor">
<a href="#genomes-managment" class="anchor"></a>Genomes managment</h1>
<p>The <code>Manage reference genomes</code> tab allows to add or remove reference genomes installed with SeqPlots. New genomes are added to user data folder, so they can be easily moved along with data and would not be removed when updating R or SeqPlots. User can choose to install standard genomes package available with Bioconductor or install forged genomic package from local file.</p>
<blockquote>
<p><img src="/Users/przemol/code/seqplots/inst/seqplots/www/help/img/11_01.png" alt="The view on Manage reference genomes panel"> Error messages ==============</p>
</blockquote>
<pre><code>Problem with line N: "line_text" [internal_error]</code></pre>
<blockquote>
<p>The import of feature file (GFF or BED) was not successful due to mis-formatted file.</p>
</blockquote>
<hr>
<pre><code>Chromosome names provided in the file does not match ones defined in reference genome. 
INPUT: [chr3R, chr2L, chr2R, chr3L] 
GENOME: [chrI, chrII, chrIII, chrIV, chrV, ...]</code></pre>
<blockquote>
<p>There are unexpected chromosome names in input file. Following genomes: <em>Arabidopsis thaliana, Caenorhabditis elegans, Cyanidioschyzon_merolae, Drosophila melanogaster, Homo sapiens, Oryza sativa, Populus trichocarpa, Saccharomyces cerevisiae and Zea mays</em> support chromosome names remapping between different naming conventions, including: AGPv2, ASM9120v1, Ensembl, JGI2_0, MSU6, NCBI, TAIR10 and UCSC. If you see above error in one of these genomes there are still unexpected names after the correction. The problematic chromosome names are given in the error message. Remove GFF/BED lines corresponding to them or upgrade the genome to one containing proper naming. Alternatively set genome to NA.</p>
</blockquote>
<hr>
<pre><code>File already exists, change the name or remove old one.</code></pre>
<blockquote>
<p>File named like this already exists in the database, it is impossible to have two files sharing same filename.</p>
</blockquote>
<hr>
<pre><code>ERROR: solving row 300: negative widths are not allowed</code></pre>
<blockquote>
<p>The the row 300 have end coordinate smaller than beginning, hence the width in negative. To fix it the start and stop indicates should be swapped. This error often happens when negative strand (-) ranges are misformatted.</p>
</blockquote>
<table style="width:24%;" class="table">
<colgroup><col width="23%"></colgroup>
<tbody><tr class="odd">
<td align="left">Explanations ============</td>
</tr></tbody>
</table>
<ul>
<li>
<p>“<strong>feature</strong>” - a genomic interval defined by <strong>chromosome</strong> name, <strong>start</strong> and <strong>end</strong> positions and the <strong>directionality</strong> (strand). The end must always be a bigger number than start, so the width of the range is not negative. Start and end means here the numeric start of the interval and should not be confused with TSS and TTS.</p>
<p>For example, in BED format this information is stored in following text tab delimited format: <code>chr7    127471196  127472363  .  .  +</code></p>
<hr>
</li>
<li>
<p>“<strong>directionality</strong>” - the strand of genomic feature, determining if the plotting range should be anchored around the star or and, and the direction in which signal is being processed to create the average track or heatmap. Unknown directionality is marked by <code>*</code> and treated as <code>+</code> for calculations.</p>
<hr>
</li>
<li>
<p>“<strong>track</strong>” - the file assigning the continuous signal (score) to genomic locations across the chromosomes. The signal usually comes from sequencing experiments, like ChIP-seq, RNA-seq, DNase-seq, MNase-seq, or from computational tools, for example nucleosome occupancy prediction, CpG density.</p>
<p>For example, in BedGraph format this information is stored in following text tab delimited format: <code>chr19 49302300 49302600 -0.75</code></p>
<hr>
</li>
<li>
<p>“<strong>reference genome package</strong>” - the R BSgemome package containing the full reference sequence for given species. It is also used to provide universal chromosome names and chromosome lengths taht are used as plotting boundaries.</p>
<hr>
</li>
<li>
<p>“<strong>reads coverage</strong>” - The basic way to calculate the signal from sequencing based assays. The numeric representation shows how much reads was aligned to given genomic location. This can be a proxy to protein-DNA binding (ChIP-seq) or the expression (RNA-seq). Can be calculated using BedTools: <a href="http://bedtools.readthedocs.org/en/latest/content/tools/genomecov.html" class="uri">http://bedtools.readthedocs.org/en/latest/content/tools/genomecov.html</a> Also known as <code>pileups</code>.</p>
<hr>
<h1 id="references">References</h1>
</li>
</ul>
<blockquote>
<p>This section lists R, JavaScript and CSS libraries used by SeqPlots, important conceptual contributions to the software, and publication, where figures generated by SeqPlots are featured.</p>
</blockquote>
<p><strong>R project and Bioconductor</strong></p>
<ul>
<li>R Core Team (2014). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL <a href="http://www.R-project.org/" class="uri">http://www.R-project.org/</a>.</li>
<li>Bioconductor: Open software development for computational biology and bioinformatics R. Gentleman, V. J. Carey, D. M. Bates, B.Bolstad, M.Dettling, S. Dudoit, B. Ellis, L. Gautier, Y. Ge, and others 2004, Genome Biology, Vol. 5, R80. URL <a href="http://www.bioconductor.org/" class="uri">http://www.bioconductor.org/</a>.</li>
<li>RStudio and Inc. (2014). shiny: Web Application Framework for R. R package version 0.10.1. <a href="http://shiny.rstudio.com/" class="uri">http://shiny.rstudio.com/</a>
</li>
<li>
<strong>Other CRAN packages:</strong> digest, DBI,RSQLite, RJSONIO, plotrix, fields, grid, kohonen, ggplot2, Cairo and parallel</li>
<li>
<strong>Bioconductor packages:</strong> IRanges, BSgenome, Rsamtools, rtracklayer, GenomicRanges and Biostrings</li>
</ul>
<p><strong>JavaScript and CSS</strong></p>
<ul>
<li>jQuery framework - <a href="http://jquery.com" class="uri">http://jquery.com</a>
</li>
<li>Bootstrap - <a href="http://getbootstrap.com" class="uri">http://getbootstrap.com</a>
</li>
<li>DataTables, Table plug-in for jQuery - <a href="http://www.datatables.net" class="uri">http://www.datatables.net</a>
</li>
<li>jQuery File Upload Plugin - <a href="https://github.com/blueimp/jQuery-File-Upload" class="uri">https://github.com/blueimp/jQuery-File-Upload</a>
</li>
<li>jQuery throttle - <a href="http://benalman.com/projects/jquery-throttle-debounce-plugin/" class="uri">http://benalman.com/projects/jquery-throttle-debounce-plugin/</a>
</li>
<li>jQuery Cookie Plugin - <a href="https://github.com/carhartl/jquery-cookie" class="uri">https://github.com/carhartl/jquery-cookie</a>
</li>
<li>Modernizer JS library - <a href="http://modernizr.com" class="uri">http://modernizr.com</a>
</li>
<li>JavaScript Templates - <a href="https://github.com/blueimp/JavaScript-Templates" class="uri">https://github.com/blueimp/JavaScript-Templates</a>
</li>
<li>JavaScript Color Picker - <a href="http://jscolor.com" class="uri">http://jscolor.com</a>
</li>
<li>md5-js - <a href="https://github.com/wbond/md5-js" class="uri">https://github.com/wbond/md5-js</a>
</li>
<li>Font Awesome - <a href="http://fortawesome.github.io/Font-Awesome" class="uri">http://fortawesome.github.io/Font-Awesome</a>
</li>
<li>Google Fonts - <a href="https://www.google.com/fonts" class="uri">https://www.google.com/fonts</a>
</li>
<li>jQuery user interface - <a href="http://jqueryui.com" class="uri">http://jqueryui.com</a> (documentation)</li>
<li>jquery.tocify.js: jQuery Table of Contents - <a href="https://github.com/gfranko/jquery.tocify.js" class="uri">https://github.com/gfranko/jquery.tocify.js</a> (documentation)</li>
<li>Strapdown <a href="https://github.com/arturadib/strapdown" class="uri">https://github.com/arturadib/strapdown</a> (documentation)</li>
<li>Bootswatch themes - <a href="http://bootswatch.com" class="uri">http://bootswatch.com</a> (documentation)</li>
<li>google-code-prettify - <a href="https://code.google.com/p/google-code-prettify" class="uri">https://code.google.com/p/google-code-prettify</a> (documentation)</li>
<li>marked - <a href="https://github.com/chjj/marked" class="uri">https://github.com/chjj/marked</a> (documentation)</li>
</ul>
<p><strong>Important conceptual contribution to the project</strong></p>
<ul>
<li>Liu T, Ortiz J, Taing L, Meyer C, Lee B, Zhang Y, Shin H, Wong S, Ma J, Lei Y, et al. 2011. <a href="http://www.ncbi.nlm.nih.gov/pubmed/21859476">Cistrome: an integrative platform for transcriptional regulation studies.</a> Genome Biology 12: R83.</li>
<li>Thomas Williams, Colin Kelley and others (2010). Gnuplot 4.4: an interactive plotting program. URL <a href="http://www.R-project.org/" class="uri">http://www.R-project.org/</a>.</li>
<li>Kent, W.J., Sugnet, C.W., Furey, T.S., Roskin, K.M., Pringle, T.H., Zahler, A.M. and Haussler, a. D. (2002). <a href="http://www.ncbi.nlm.nih.gov/pubmed/12045153">The Human Genome Browser at UCSC</a>. Genome Research. 12:996–1006.</li>
<li>Kent WJ, Zweig AS, Barber G, Hinrichs AS, Karolchik D. (2010). <a href="http://www.ncbi.nlm.nih.gov/pubmed/20639541">BigWig and BigBed: enabling browsing of large distributed datasets.</a> Bioinformatics. 1;26(17):2204-7</li>
<li>Nicol, J.W., Helt, G.A., Blanchard, S.G., Raja, A. and Loraine, A.E. (2009). <a href="http://www.ncbi.nlm.nih.gov/pubmed/19654113">The Integrated Genome Browser: free software for distribution and exploration of genome-scale datasets.</a> Bioinformatics (Oxford, England). 25:2730–1.</li>
<li>Thorvaldsdóttir, H., Robinson, J.T. and Mesirov, J.P. (2012). <a href="http://www.ncbi.nlm.nih.gov/pubmed/22517427">Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration.</a> Briefings in bioinformatics. bbs017</li>
</ul>
<p><strong>Server deployment</strong></p>
<ul>
<li>Shiny Server - <a href="https://github.com/rstudio/shiny-server" class="uri">https://github.com/rstudio/shiny-server</a>
</li>
<li>ShinyApps - <a href="https://github.com/rstudio/shinyapps" class="uri">https://github.com/rstudio/shinyapps</a>
</li>
</ul>
<p><strong>Publications containing figures made by SeqPlots</strong></p>
<ul>
<li>Chen RA, Stempor P, Down TA, Zeiser E, Feuer SK, Ahringer J. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24653213">Extreme HOT regions are CpG-dense promoters in C. elegans and humans.</a> Genome Res 24(7):1138-1146 Jul 2014</li>
<li>Latorre I, Chesney MA, Garrigues JM, Stempor P et al. <a href="http://www.ncbi.nlm.nih.gov/pubmed/25737279">The DREAM complex promotes gene body H2A.Z for target repression</a>. Genes Dev 2015 Mar 1;29(5):495-500.</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<pre><code>## R version 3.4.0 (2017-04-21)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.5
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] compiler_3.4.0  backports_1.1.0 magrittr_1.5    rprojroot_1.2  
##  [5] tools_3.4.0     htmltools_0.3.6 yaml_2.1.14     Rcpp_0.12.12   
##  [9] stringi_1.1.5   rmarkdown_1.6   knitr_1.16      stringr_1.2.0  
## [13] digest_0.6.12   evaluate_0.10.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#adding-and-managing-files">Adding and managing files</a><ul class="nav nav-pills nav-stacked">
<li><a href="#supported-file-formats">Supported file formats</a></li>
      <li><a href="#adding-files">Adding files</a></li>
      <li><a href="#downloading-and-removing-files">Downloading and removing files</a></li>
      </ul>
</li>
      <li>
<a href="#running-the-plot-set-jobs">Running the plot-set jobs</a><ul class="nav nav-pills nav-stacked">
<li><a href="#selecting-files">Selecting files</a></li>
      <li><a href="#setting-up-plot-options">Setting up plot options</a></li>
      <li><a href="#plotting-sequence-motif-density">Plotting sequence motif density</a></li>
      <li><a href="#starting-the-plot-set-calculation">Starting the plot set calculation</a></li>
      </ul>
</li>
      <li>
<a href="#plotting">Plotting</a><ul class="nav nav-pills nav-stacked">
<li><a href="#previewing-plot">Previewing plot</a></li>
      <li><a href="#titles-and-axis-panel">Titles and axis panel</a></li>
      <li><a href="#guide-lines-and-data-scaling">Guide lines and data scaling</a></li>
      <li><a href="#keys-labels-and-colors-panel">Keys, labels and colors panel</a></li>
      <li><a href="#heatmap-setup-tab">Heatmap setup tab</a></li>
      <li><a href="#other-options-controlling-heatmap-appearance">Other options controlling heatmap appearance</a></li>
      </ul>
</li>
      <li>
<a href="#getting-pdfs-and-cluster-info">Getting PDFs and cluster info</a><ul class="nav nav-pills nav-stacked">
<li><a href="#pdf-output-size">PDF output size</a></li>
      </ul>
</li>
      <li>
<a href="#batch-operations">Batch operations</a><ul class="nav nav-pills nav-stacked">
<li><a href="#load-or-save-plotset">Load or save plotset</a></li>
      <li><a href="#plot-set-files-structure">Plot set files structure</a></li>
      </ul>
</li>
      <li><a href="#advanced-options">Advanced options</a></li>
      <li><a href="#genomes-managment">Genomes managment</a></li>
      <li><a href="#session-information">Session Information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
